### PR TITLE
Add support for nested requests in MetricsClientHttpRequestInterceptor

### PR DESCRIPTION
**Pull request by nosan**
See gh-19381

### CODE REVIEW

File: spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/web/client/MetricsClientHttpRequestInterceptor.java [hunk 1]
```diff
@@ -96,7 +98,9 @@ public ClientHttpResponse intercept(HttpRequest request, byte[] body, ClientHttp
 		finally {
 			getTimeBuilder(request, response).register(this.meterRegistry).record(System.nanoTime() - startTime,
 					TimeUnit.NANOSECONDS);
-			urlTemplate.remove();
+			if (urlTemplate.get().isEmpty()) {
+				urlTemplate.remove();
+			}
```
Comment (grahammkelly) [inline spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/web/client/MetricsClientHttpRequestInterceptor.java]: How does the previous value get popped? Should this not have 

```java
... 
} else {
   urlTemplate.get.pop()
}
```

**Comment by grahammkelly** [comment]
Looks good and it addresses the issue.

**Comment by grahammkelly** [comment]
Will/can this be backported to the 2.1 branch?

**Comment by wilkinsona** [comment]
We can certainly consider it. IIRC, we only made the change in 2.3.x as we were concerned about the risk of a regression.

**Pull request closed without merging**